File: GuardianMonsters/battle/src/de/limbusdev/guardianmonsters/battle/ui/widgets/ObservableWidget.java
package de.limbusdev.guardianmonsters.battle.ui.widgets;

/**
 * When implementing ObservableWidget, prevent observers registering multiple times
 * Created by georg on 17.08.16.
 */
public interface ObservableWidget {
    void addWidgetObserver(WidgetObserver wo);
    void notifyWidgetObservers();
}

File: GuardianMonsters/battle/src/de/limbusdev/guardianmonsters/battle/ui/widgets/ObservableWidget.kt
package de.limbusdev.guardianmonsters.battle.ui.widgets

/**
 * When implementing ObservableWidget, prevent observers registering multiple times
 *
 * @author Georg Eckert 2019
 */
interface ObservableWidget
{
    fun addWidgetObserver(wo: WidgetObserver)
    fun notifyWidgetObservers()
}
